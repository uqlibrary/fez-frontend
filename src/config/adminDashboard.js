import React from 'react';
import { default as adminDashboardLocale } from 'locale/components';

import Chip from '@mui/material/Chip';

import Today from 'modules/AdminDashboard/tabs/Today';
import SystemAlerts from 'modules/AdminDashboard/tabs/SystemAlerts';
import Reports from 'modules/AdminDashboard/tabs/Reports';

export const tabConfig = [
    { id: 0, title: adminDashboardLocale.components.adminDashboard.tabs.today.tabLabel, component: Today },
    {
        id: 1,
        title: adminDashboardLocale.components.adminDashboard.tabs.systemalerts.tabLabel,
        component: SystemAlerts,
    },
    { id: 2, title: adminDashboardLocale.components.adminDashboard.tabs.reports.tabLabel, component: Reports },
];

export const tabProps = [
    {
        id: 1,
        render: data => {
            const count = data?.systemalerts?.mine;
            return count
                ? {
                      icon: <Chip color="error" label={count} size="small" data-testid={`tab-counter-${count}`} />,
                      iconPosition: 'end',
                  }
                : {};
        },
    },
];
