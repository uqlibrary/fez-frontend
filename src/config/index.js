export {
    API_URL,
    APP_URL,
    AUTH_URL_LOGIN,
    AUTH_URL_LOGOUT,
    SESSION_COOKIE_NAME,
    SESSION_USER_GROUP_COOKIE_NAME,
    TOKEN_NAME,
    ORCID_BASE_URL,
    ORCID_CLIENT_ID,
    ORCID_AUTHORIZATION_URL,
    GOOGLE_MAPS_API_URL,
    GOOGLE_MAPS_API_CHINA_URL,
    PATH_PREFIX,
    DATASET_ACCESS_CONDITIONS_OPTIONS,
} from './general';

export { api, generateCancelToken, sessionApi } from './axios';
export * as validation from './validation';
export * as general from './general';
export { publicationTypes, contentIndicators, numberToWords } from './general';
export * as routes from './routes';
export * as openAccessConfig from './openAccess';
export { default as incompleteRecord } from './incompleteRecord';
export { viewRecordsConfig } from './viewRecord';
export { trendingPublicationsConfig } from './trendingPublications';
export { claimRecordConfig } from './claimRecord';
export { mui1theme, adminTheme } from './theme';
export * as BULK_UPDATES from './bulkUpdates';
export { pathConfig } from './pathConfig';
export { viewJournalConfig } from './viewJournal';
export { TRANSITION_COHORT } from './thesisSubmissionTransition';

export { default as imageGalleryConfig } from './imageGalleryConfig';
export { communityCollectionsConfig } from './communityCollections';
export { controlledVocabConfig } from './controlledVocabConfig';
