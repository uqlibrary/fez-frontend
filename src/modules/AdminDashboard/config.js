import React from 'react';
import moment from 'moment';

import Chip from '@mui/material/Chip';
import Typography from '@mui/material/Typography';

import { ExternalLink } from 'modules/SharedComponents/ExternalLink';

export const COLOURS = { assigned: '#338CFA', unassigned: '#B60DCE' };

export const LINK_UNPROCESSED_WORKS =
    'https://espace.library.uq.edu.au/records/search?page=1&pageSize=20&sortBy=score&sortDirection=Desc&searchQueryParams%5Ball%5D%5Bvalue%5D=ismemberof%3A+UQ%3A218198&searchQueryParams%5Ball%5D%5Blabel%5D=&searchMode=advanced';

export const animationTemplate = (i, duration, delay) =>
    `animateFadeIn ${duration}ms ease-out ${delay * (i + 1)}ms forwards`;

export const MENUACTIONS = {
    EDIT: 'EDIT',
    DELETE: 'DELETE',
    UP: 'MOVEUP',
    TOP: 'MOVETOP',
    DOWN: 'MOVEDOWN',
    BOTTOM: 'MOVEBOTTOM',
};

export const VIEWMODES = {
    VIEW: 'VIEW',
    ADD: 'ADD',
    EDIT: 'EDIT',
    DELETE: 'DELETE',
};
export const VIEWADMINPANELMODES = [VIEWMODES.ADD, VIEWMODES.EDIT, VIEWMODES.DELETE];
export const REORDERING = [MENUACTIONS.TOP, MENUACTIONS.UP, MENUACTIONS.BOTTOM, MENUACTIONS.DOWN];

export const DEFAULT_DATEPICKER_INPUT_FORMAT = 'DD/MM/YYYY';
export const DEFAULT_DATE_FORMAT = 'Do MMMM YYYY';
export const DEFAULT_DATE_FORMAT_WITH_TIME = 'Do MMMM YYYY hh:mm';
export const DEFAULT_DATE_FORMAT_WITH_TIME_24H = 'Do MMMM YYYY HH:mm';
export const DEFAULT_DATE_FORMAT_WITH_TIME_24H_SECONDS = 'Do MMMM YYYY HH:mm:ss';
export const DEFAULT_SERVER_DATE_FORMAT = 'YYYY-MM-DD HH:mm:ss';

export const SYSTEM_ALERT_ACTION = {
    ASSIGN: 'ASSIGN',
    RESOLVE: 'RESOLVE',
};

export const REPORT_TYPE = {
    systemalertlog: 1,
    workshistory: 2,
    workiddupe: 3,
    scopusiddupe: 4,
    doidupe: 5,
};

export const DEFAULT_SORTING = {
    alerts: [{ field: 'sat_created_date', sort: 'asc' }],
    systemalertlog: [{ field: 'sat_created_date', sort: 'asc' }],
    workshistory: [{ field: 'pre_date', sort: 'asc' }],
};

export const isUrl = str => {
    try {
        const newUrl = new URL(str);
        return newUrl.protocol === 'http:' || newUrl.protocol === 'https:';
    } catch (err) {
        return false;
    }
};

export const defaultLegacyReportOption = { sel_id: 0, sel_title: '', sel_description: '' };

export const optionDoubleRowRender = (props, option) => {
    return (
        <li
            {...props}
            style={{
                flexDirection: 'column',
                justifyContent: 'left',
                alignItems: 'flex-start',
                fontWeight: 400,
            }}
            data-testid={props.id}
        >
            <Typography variant="body1" color="textPrimary">
                {option.sel_title}
            </Typography>
            <Typography variant="body1" color="textSecondary">
                {option.sel_description}
            </Typography>
        </li>
    );
};

export const getReportTypeFromValue = value => Object.entries(REPORT_TYPE).find(arr => arr[1] === value)?.[0];
export const getDefaultSorting = reportType => DEFAULT_SORTING?.[reportType] || [];

export const getFormattedServerDate = (dateStr, format = DEFAULT_DATE_FORMAT) =>
    (dateStr && moment(dateStr, DEFAULT_SERVER_DATE_FORMAT).format(format)) || '';

export const getSystemAlertColumns = (locale, users) => {
    const alertStatus = locale.alertStatus;
    const alertStatusOption = Object.values(alertStatus);
    return [
        {
            field: 'sat_created_date',
            headerName: locale.columns.createdDate,
            width: 200,
            type: 'date',
            valueGetter: value => moment(value, DEFAULT_SERVER_DATE_FORMAT).toDate(),
            valueFormatter: value => moment(value).format(DEFAULT_DATE_FORMAT_WITH_TIME_24H),
        },
        { field: 'sat_title', headerName: locale.columns.topic, flex: 1 },
        {
            field: 'status',
            headerName: locale.columns.status,
            width: 160,
            valueGetter: (_, row) =>
                !!row.sat_assigned_to
                    ? users.find(user => user.id === row.sat_assigned_to)?.preferred_name ?? alertStatus.UNKNOWN
                    : alertStatus.UNASSIGNED,
            renderCell: params => (
                <Chip
                    data-testid={`alert-status-${params.id}`}
                    label={params.value}
                    variant="filled"
                    size="small"
                    color={alertStatusOption.includes(params.value) ? 'default' : 'primary'}
                />
            ),
        },
    ];
};

export const getDisplayReportColumns = ({ locale, actionState, params }) => {
    const report = actionState?.displayReport?.value || getReportTypeFromValue(params.report_type);

    const txt = locale.columns[report];
    switch (report) {
        case 'workshistory':
            return [
                { field: 'pre_id', headerName: txt.id, width: 150, exportOnly: true, exportOrder: 0 },
                { field: 'pre_pid', headerName: txt.pid, width: 150, order: 0, exportOrder: 1 },
                {
                    field: 'pre_date',
                    headerName: txt.dateCreated,
                    width: 150,
                    type: 'date',
                    valueGetter: value => (!!value && moment(value, DEFAULT_SERVER_DATE_FORMAT).toDate()) || null,
                    valueFormatter: value => {
                        if (value === null || value === '') return '';
                        return moment(value).format(DEFAULT_DATE_FORMAT);
                    },
                    order: 1,
                    exportOrder: 5,
                },
                {
                    field: 'rek_date',
                    headerName: txt.pubDate,
                    width: 150,
                    type: 'date',
                    valueGetter: value =>
                        (!!value && moment(value, DEFAULT_SERVER_DATE_FORMAT).format(DEFAULT_DATE_FORMAT)) || '',
                    exportOnly: true,
                    exportOrder: 4,
                },
                { field: 'rek_genre', headerName: txt.genre, minWidth: 300, flex: 1, order: 2, exportOrder: 2 },
                { field: 'rek_subtype', headerName: txt.subtype, minWidth: 300, flex: 1, order: 3, exportOrder: 3 },
                { field: 'usr_username', headerName: txt.user, width: 150, order: 4, exportOrder: 6 },
                { field: 'pre_detail', headerName: txt.action, minWidth: 600, flex: 1, order: 5, exportOrder: 7 },
            ];
        default:
            const systemIdParam = actionState?.systemAlertId || params?.record_id || '';
            const cols = [
                { field: 'sat_id', headerName: txt.id, order: 0, exportOrder: 0 },
                {
                    field: 'sat_created_date',
                    headerName: txt.dateCreated,
                    width: 150,
                    type: 'date',
                    valueGetter: value => (!!value && moment(value, DEFAULT_SERVER_DATE_FORMAT).toDate()) || null,
                    valueFormatter: value => {
                        if (value === null || value === '') return '';
                        return moment(value).format(DEFAULT_DATE_FORMAT);
                    },
                    order: 1,
                    exportOrder: 1,
                },
                {
                    field: 'assigned_to_username',
                    headerName: 'Assigned username',
                    width: 150,
                    exportOrder: 2,
                    exportOnly: true,
                },
                {
                    field: 'assigned_to_full_name',
                    headerName: txt.assignedTo,
                    width: 150,
                    exportOrder: 3,
                    order: 2,
                },
                {
                    field: 'sat_assigned_date',
                    headerName: txt.assignedDate,
                    width: 150,
                    type: 'date',
                    valueGetter: value => (!!value && moment(value, DEFAULT_SERVER_DATE_FORMAT).toDate()) || null,
                    valueFormatter: value => {
                        if (value === null || value === '') return '';
                        return moment(value).format(DEFAULT_DATE_FORMAT);
                    },
                    order: 3,
                    exportOrder: 4,
                },
                {
                    field: 'resolved_by_username',
                    headerName: 'Resolved username',
                    width: 150,
                    exportOrder: 5,
                    exportOnly: true,
                },
                {
                    field: 'resolved_by_full_name',
                    headerName: txt.resolvedBy,
                    width: 150,
                    order: 4,
                    exportOrder: 6,
                },
                {
                    field: 'sat_resolved_date',
                    headerName: txt.resolvedDate,
                    width: 150,
                    type: 'date',
                    valueGetter: value => (!!value && moment(value, DEFAULT_SERVER_DATE_FORMAT).toDate()) || null,
                    valueFormatter: value => {
                        if (value === null || value === '') return '';
                        return moment(value).format(DEFAULT_DATE_FORMAT);
                    },
                    order: 5,
                    exportOrder: 7,
                },
                { field: 'sat_title', headerName: txt.title, minWidth: 400, flex: 1, order: 6, exportOrder: 8 },
                {
                    field: 'sat_link',
                    headerName: txt.link,
                    minWidth: 400,
                    flex: 1,
                    renderCell: row =>
                        isUrl(row.value) ? (
                            <ExternalLink id={`link_${row.id}`} href={row.value} inline>
                                {row.value}
                            </ExternalLink>
                        ) : (
                            row.value
                        ),
                    order: 8,
                    exportOrder: 9,
                },
            ];

            if (systemIdParam !== '') {
                cols.push({
                    field: 'sat_content',
                    headerName: txt.content,
                    minWidth: 1000,
                    flex: 1,
                    order: 7,
                    exportOrder: 10,
                });
            }
            return cols;
    }
};
