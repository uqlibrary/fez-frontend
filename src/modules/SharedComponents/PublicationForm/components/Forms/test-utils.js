import { useForm } from 'react-hook-form';
import { render, WithReduxStore } from 'test-utils';
import React from 'react';
import { formValues } from '../PublicationForm';

export const setupWrapper = (Form, props) => {
    const Wrapper = () => {
        const {
            control,
            formState: { isSubmitting },
        } = useForm();

        // eslint-disable-next-line react/prop-types
        return <Form {...{ control, isSubmitting, ...props, formValues: formValues(props?.formValues || {}) }} />;
    };

    return render(
        <WithReduxStore>
            <Wrapper />
        </WithReduxStore>,
    );
};
