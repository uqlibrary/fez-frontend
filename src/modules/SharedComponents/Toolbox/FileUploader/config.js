export const FILE_ACCESS_CONDITION_CLOSED = 1;
export const FILE_ACCESS_CONDITION_OPEN = 5;
export const FILE_ACCESS_CONDITION_INHERIT = 99;
export const FILE_SECURITY_POLICY_ADMIN = 1;
export const FILE_SECURITY_POLICY_EVIDENCE_ASSESSORS = 3;
export const FILE_SECURITY_POLICY_PUBLIC = 5;

export const FILE_ACCESS_OPTIONS = [
    { text: 'Open Access', value: FILE_ACCESS_CONDITION_OPEN },
    { text: 'Closed Access', value: FILE_ACCESS_CONDITION_CLOSED },
];

export const INHERIT_OPTION = {
    text: 'Inherit from parent collection',
    value: FILE_ACCESS_CONDITION_INHERIT,
};

export const SIZE_BASE = 1000;

export const DEFAULT_FILE_UPLOAD_LIMIT = 10;
export const DEFAULT_MAX_FILE_SIZE = 5;

export const SIZE_UNIT_B = 'B';
export const SIZE_UNIT_K = 'K';
export const SIZE_UNIT_M = 'M';
export const SIZE_UNIT_G = 'G';

export const FILE_META_KEY_ACCESS_CONDITION = 'access_condition_id';
export const FILE_META_KEY_EMBARGO_DATE = 'date';
export const FILE_META_KEY_DESCRIPTION = 'description';
export const FILE_META_KEY_SECURITY_POLICY = 'security_policy';

export const SIZE_UNITS = [SIZE_UNIT_B, SIZE_UNIT_K, SIZE_UNIT_M, SIZE_UNIT_G];

export const FILE_NAME_RESTRICTION = /^(?=^\S*$)(?=^[a-z\d\-_]+(\.\d{3}|\.r\d{2,3}|\.part\d{1,3})?\.[^\.]+$)(?=.{1,45}$)(?!(web_|preview_|thumbnail_|stream_|fezacml_|presmd_|\d))[a-z\d\-_\.]+/;

/**
 * When updating the following list, please make sure to update it as well in the following projects:
 * - https://github.com/uqlibrary/fez-frontend
 * - https://github.com/uqlibrary/lambda-fez-upload-handler-trigger
 * - https://github.com/uqlibrary/api
 * - https://github.com/uqlibrary/fez
 *
 * Note: ext -> mimetype is a 1-n relation, but we only care about one value
 */
export const MIME_TYPE_WHITELIST = {
    pdf: 'application/pdf',
    jpe: 'image/jpeg',
    jpeg: 'image/jpeg',
    jpg: 'image/jpeg',
    gif: 'image/gif',
    tif: 'image/tiff',
    tiff: 'image/tiff',
    png: 'image/png',
    zip: 'application/zip',
    tar: 'application/x-tar',
    gz: 'application/gzip',
    '7z': 'application/x-7z-compressed',
    rar: 'application/x-rar-compressed',
    mk3d: 'video/x-matroska',
    mks: 'video/x-matroska',
    mkv: 'video/x-matroska',
    qt: 'video/quicktime',
    mov: 'video/quicktime',
    mp4: 'video/mp4',
    mp4v: 'video/mp4',
    mp3: 'audio/mp3',
    mxf: 'application/mxf',
    wav: 'audio/wav',
    mpe: 'video/mpeg',
    mpeg: 'video/mpeg',
    mpg: 'video/mpeg',
    m1v: 'video/mpeg',
    m2v: 'video/mpeg',
    wmv: 'video/x-ms-wmv',
    wma: 'audio/x-ms-wma',
    avi: 'video/x-msvideo',
    m4a: 'audio/x-m4a',
    csv: 'text/csv',
    xla: 'application/vnd.ms-excel',
    xlc: 'application/vnd.ms-excel',
    xlm: 'application/vnd.ms-excel',
    xls: 'application/vnd.ms-excel',
    xlt: 'application/vnd.ms-excel',
    xlw: 'application/vnd.ms-excel',
    xlsx: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
    ods: 'application/vnd.oasis.opendocument.spreadsheet',
    gsheet: 'application/vnd.google-apps.spreadsheet',
};
