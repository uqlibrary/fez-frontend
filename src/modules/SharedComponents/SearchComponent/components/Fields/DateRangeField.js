import React, { useState, useEffect, useCallback } from 'react';
import PropTypes from 'prop-types';
import TextField from '@mui/material/TextField';
import Grid from '@mui/material/Grid';
import InputLabel from '@mui/material/InputLabel';

import { AdapterMoment } from '@mui/x-date-pickers/AdapterMoment';
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
import { DatePicker } from '@mui/x-date-pickers/DatePicker';
import { GENERIC_DATE_FORMAT } from 'config/general';

export const DateRangeField = ({
    id,
    locale = {},
    format = GENERIC_DATE_FORMAT,
    from: initialFrom,
    to: initialTo,
    onChange,
    disableFuture = false,
    disabled = false,
}) => {
    const [from, setFrom] = useState(initialFrom || null);
    const [to, setTo] = useState(initialTo || null);
    const [error, setError] = useState(undefined);
    const [fromError, setFromError] = useState(undefined);
    const [toError, setToError] = useState(undefined);
    const [allGood, setAllGood] = useState(false);

    useEffect(() => {
        if (!!from && !from.isValid()) {
            setFromError('Please provide valid date');
        } else if (!!to && !to.isValid()) {
            setToError('Please provide valid date');
        } else if (!!from && !!to && from.isAfter(to)) {
            setError('Please provide valid date range');
        } else {
            setFromError(undefined);
            setToError(undefined);
            setError(undefined);
        }
        setAllGood(true);
    }, [from, to]);

    useEffect(() => {
        if (allGood && !error && !!from && !!to) {
            onChange({ from, to });
            setAllGood(false);
        }
    }, [allGood, error, from, fromError, onChange, to, toError]);

    const handleFromDateChange = useCallback(value => {
        setFrom(value);
    }, []);

    const handleToDateChange = useCallback(value => {
        setTo(value);
    }, []);

    return (
        <React.Fragment>
            <Grid container>
                <Grid item xs={12}>
                    <InputLabel shrink sx={theme => ({ ...theme.typography.caption })}>
                        {locale.title}
                    </InputLabel>
                </Grid>
            </Grid>
            <Grid container>
                <Grid item zeroMinWidth sx={{ flexGrow: 1, width: '1px' }}>
                    <LocalizationProvider dateAdapter={AdapterMoment}>
                        <DatePicker
                            value={from}
                            onChange={handleFromDateChange}
                            disableToolbar
                            inputFormat={format}
                            inputProps={{ 'data-testid': `${id}-from-date` }}
                            disableFuture={disableFuture}
                            disabled={disabled}
                            renderInput={params => (
                                <TextField
                                    {...params}
                                    id={`${id}-from-date`}
                                    error={!!error || !!fromError}
                                    helperText={error || fromError}
                                    variant="standard"
                                />
                            )}
                        />
                    </LocalizationProvider>
                </Grid>
                <Grid item xs="auto">
                    <TextField
                        variant="standard"
                        sx={{ width: '24px' }}
                        value=" to "
                        disabled
                        InputProps={{ disableUnderline: true }}
                    />
                </Grid>
                <Grid item zeroMinWidth sx={{ flexGrow: 1, width: '1px' }}>
                    <LocalizationProvider dateAdapter={AdapterMoment}>
                        <DatePicker
                            value={to}
                            onChange={handleToDateChange}
                            disableToolbar
                            inputFormat={format}
                            inputProps={{ 'data-testid': `${id}-to-date` }}
                            disableFuture={disableFuture}
                            disabled={disabled}
                            renderInput={params => (
                                <TextField
                                    {...params}
                                    id={`${id}-to-date`}
                                    error={!!error || !!toError}
                                    helperText={toError}
                                    variant="standard"
                                />
                            )}
                        />
                    </LocalizationProvider>
                </Grid>
            </Grid>
        </React.Fragment>
    );
};

DateRangeField.propTypes = {
    id: PropTypes.string,
    locale: PropTypes.object,
    format: PropTypes.string,
    from: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.object]),
    to: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.object]),
    onChange: PropTypes.func.isRequired,
    disableFuture: PropTypes.bool,
    disabled: PropTypes.bool,
};

export default React.memo(DateRangeField);
