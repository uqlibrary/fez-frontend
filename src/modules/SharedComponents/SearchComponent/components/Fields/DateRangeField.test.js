import React from 'react';
import DateRangeField from './DateRangeField';
import { GENERIC_DATE_FORMAT } from 'config/general';
import { rtlRender, fireEvent, act } from 'test-utils';

function setup(testProps = {}) {
    const props = {
        disabled: false,
        locale: {},
        format: GENERIC_DATE_FORMAT,
        from: '',
        to: '',
        onChange: jest.fn(),
        disableFuture: false,
        ...testProps,
    };

    return rtlRender(<DateRangeField {...props} />);
}

describe('DateRangeField component', () => {
    it('should render with default props', () => {
        const { container } = rtlRender(<DateRangeField onChange={jest.fn()} />);
        expect(container).toMatchSnapshot();
    });

    it('should render default view', () => {
        const { getByTestId } = setup({ id: 'test-date-range' });
        expect(getByTestId('test-date-range-from-date')).toBeInTheDocument();
        expect(getByTestId('test-date-range-to-date')).toBeInTheDocument();
    });

    it('should render disabled view', () => {
        const { getByTestId } = setup({ disabled: true, id: 'test-date-range' });
        expect(getByTestId('test-date-range-from-date').disabled).toBeTruthy();
        expect(getByTestId('test-date-range-to-date')).toBeTruthy();
    });

    it('should display error for "from" date for invalid date input', () => {
        const { getByTestId, getByText } = setup({ id: 'test-date-range' });
        fireEvent.change(getByTestId('test-date-range-from-date'), { target: { value: '10/13/2010' } });
        expect(getByText('Please provide valid date')).toBeInTheDocument();
    });

    it('should display error for "to" date for invalid date input', () => {
        const { getByTestId, getByText } = setup({ id: 'test-date-range' });
        fireEvent.change(getByTestId('test-date-range-to-date'), { target: { value: '10/13/2010' } });
        expect(getByText('Please provide valid date')).toBeInTheDocument();
    });

    it('should display error for date range if both dates are valid but invalid range', () => {
        const { getByTestId, getByText } = setup({ id: 'test-date-range' });

        act(() => {
            fireEvent.change(getByTestId('test-date-range-from-date'), { target: { value: '10/10/2010' } });
        });

        act(() => {
            fireEvent.change(getByTestId('test-date-range-to-date'), { target: { value: '10/09/2010' } });
        });

        expect(getByText('Please provide valid date range')).toBeInTheDocument();
    });

    it('should not call onChange callback if one of the values (from/to) is missing', () => {
        const onChangeFn = jest.fn();
        const { getByTestId } = setup({ id: 'test-date-range', onChange: onChangeFn });
        fireEvent.change(getByTestId('test-date-range-from-date'), { target: { value: '10/10/2010' } });
        expect(onChangeFn).not.toBeCalled();
    });

    it('should call onChange once both date are entered in the range', () => {
        const onChangeFn = jest.fn();
        const { getByTestId } = setup({ id: 'test-date-range', onChange: onChangeFn });
        act(() => {
            fireEvent.change(getByTestId('test-date-range-from-date'), { target: { value: '10/10/2010' } });
        });

        act(() => {
            fireEvent.change(getByTestId('test-date-range-to-date'), { target: { value: '10/11/2010' } });
        });

        expect(onChangeFn).toHaveBeenCalled();
    });
});
