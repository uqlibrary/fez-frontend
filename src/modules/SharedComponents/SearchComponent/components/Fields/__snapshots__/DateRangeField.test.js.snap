// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangeField component should render with default props 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated MuiInputLabel-shrink css-15a1x70-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
      />
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth css-qheo1o-MuiGrid-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1gtyszw-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
            data-testid="undefined-from-date"
            id="undefined-from-date"
            placeholder="dd/mm/yyyy"
            type="tel"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <button
              aria-label="Choose date"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1d2b2o6-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CalendarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-o3gf2k-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl css-ydg547-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input Mui-disabled css-1x51dt5-MuiInputBase-input-MuiInput-input"
            disabled=""
            id=":r1:"
            type="text"
            value=" to "
          />
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth css-qheo1o-MuiGrid-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1gtyszw-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
            data-testid="undefined-to-date"
            id="undefined-to-date"
            placeholder="dd/mm/yyyy"
            type="tel"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <button
              aria-label="Choose date"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1d2b2o6-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CalendarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
